doc/apiv3-documentation.apib

```diff
@@ -3068,22 +3107,238 @@ For more details and all possible responses see the general specification of [Fo
+                                "title": "Jim Sheppard - j.sheppard2"
+                            },
+                            "lock": {
+                                "href": "/api/v3/users/2/lock",
+                                "title": "Set lock on j.sheppard2"
+                                "method": "POST"
+                            },
+                            "delete": {
+                                "href": "/api/v3/users/2",
+                                "title": "Delete j.sheppard2"
+                                "method": "DELETE"
+                            }
+                        },
+                        "id": 2,
+                        "login": "j.sheppard2",
+                        "firstName": "Jim",
+                        "lastName": "Sheppard",
+                        "mail": "shep@mail.net",
+                        "avatar": "https://gravatar/avatar",
+                        "status": "active",
+                        "createdAt": "2014-05-21T08:51:20Z",
+                        "updatedAt": "2014-05-21T08:51:20Z"
+                    }]
+                }
+            }
+
+## List watchers [GET]
+
++ Parameters
+    + work_package_id (required, integer, `1`) ... Work package id
+
++ Response 200 (application/hal+json)
+
+    [Watchers][]
+
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:** view work package watchers
+
+    *Note that you will only receive this error, if you are at least allowed to see the corresponding work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to see the watchers of this work package."
+            }
+
++ Response 404 (application/hal+json)
+
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** view work package
+
+    *Note that you will effectively not be able to see the watchers of a work package without being able to see the work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:NotFound",
+                "message": "The specified work package does not exist."
+            }
 
 ## Add watcher [POST]
 
+Adds a watcher to the specified work package.
+
+The request is expected to contain a single JSON object, that contains a link object under the `user` key.
+
+The response will be the collection of *all* the watchers for the given work package.
+
 + Parameters
     + work_package_id (required, integer, `1`) ... Work package id
 
 + Request (application/json)
 
+        {
+            "user": { "href": "/api/v3/users/1" }
+        }
+
++ Response 200 (application/hal+json)
+
+    [Watchers][]
```

Comment:
We defined to return the collection of current watchers in response to a successful add watcher call.

However, I just realized that to add a watcher it suffices to have the permission *view work package* suffices (for self). But to list watchers of a work package, it is neccessary to have to *view work package watchers* permission.

This is a conflict, because by adding self to the watchers I can effectively list all other watchers.

@ulferts Any thoughts? I'd currently say we simply return the user that was just added (current behaviour)